File: agents/java-agent/src/main/java/com/hmdev/messaging/agent/api/ConnectionChannelApiFactory.java
package com.hmdev.messaging.agent.api;



import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import com.hmdev.messaging.agent.api.http.HTTPChannelApi;



public abstract class ConnectionChannelApiFactory {
    private static final Logger logger = LoggerFactory.getLogger(ConnectionChannelApiFactory.class);

	
	private ConnectionChannelApiFactory(){
		super();
	}
	
	public static ConnectionChannelApi getConnectionApi(String descriptor) {
		if(descriptor.matches("^https?://.*")){
			return new HTTPChannelApi(descriptor);
		}else{
			throw new RuntimeException("Connection channel descriptor is not supported");
		}
	}
}

File: agents/python-agent/hmdev/messaging/agent/api/connection_channel_api_factory.py
import re
from hmdev.messaging.agent.api.http.http_channel_api import HTTPChannelApi


class ConnectionChannelApiFactory:
    @staticmethod
    def get_connection_api(remote_url: str) -> HTTPChannelApi:
        if re.match(r"^https?://", remote_url):
            return HTTPChannelApi(remote_url)
        else:
            raise RuntimeError("Connection channel descriptor is not supported")
